src/components/alert/alert.js

```diff
@@ -37,71 +36,67 @@ export default {
   data() {
     return {
       countDownTimerId: null,
-      dismissed: false,
-      localShow: this.show,
+      countDown: 0,
+      // If initially shown, we need to set these for SSR
+      localShow: Boolean(this.show),
       showClass: this.fade && this.show
     }
   },
   watch: {
     show(newVal) {
-      this.showChanged(newVal)
+      this.countDown = typeof newVal === 'number' && newVal > 0 ? newVal : 0
+      this.localShow = Boolean(newVal)
     },
-    dismissed(newVal) {
-      if (newVal) {
+    countDown(newVal) {
+      this.clearTimer()
+      this.$emit('dismiss-count-down', newVal)
+      if (this.show !== newVal) {
+        // Update the v-model if needed
+        this.$emit('input', newVal)
+      }
+      if (newVal > 0) {
+        this.localShow = true
+        this.countDownTimerId = setTimeout(() => {
+          this.countDown--
+        }, 1000)
+      } else {
         this.localShow = false
+      }
+    },
+    localShow(newVal) {
+      if (!newVal && this.dismissible) {
         this.$emit('dismissed')
       }
+      if (typeof this.show !== 'number' && this.show !== newVal) {
+        this.$emit('input', newVal)
+      }
     }
   },
+  created() {
+    this.localShow = Boolean(this.show)
+    this.countDown = typeof this.show === 'number' ? this.show : 0
```

Comment:
What about a `parseCountDown()` method?
```js
parseCountDown(value) {
  return typeof value === 'number' && value > 0 ? value : 0
}
```

Reply:
Yeah that might be a good idea...

Reply:
And maybe a `parseShow()` method too:
```js
parseShow(value) {
  return Boolean(value)
}
```

Reply:
I might make `show` accept a string number as well (but will need to test for `show === ''` as well for boolean `true`)